src/internal/client.ts

```diff
@@ -2290,4 +2291,52 @@ export class TypedClient {
     const body = await readAsBuffer(res)
     return parseSelectObjectContentResponse(body)
   }
+
+  private async applyBucketLifecycle(bucketName: string, policyConfig: LifecycleConfig): Promise<void> {
+    const method = 'PUT'
+    const query = 'lifecycle'
+
+    const headers: RequestHeaders = {}
+    const builder = new xml2js.Builder({
+      rootName: 'LifecycleConfiguration',
+      headless: true,
+      renderOpts: { pretty: false },
+    })
+    const payload = builder.buildObject(policyConfig)
+    headers['Content-MD5'] = toMd5(payload)
+
+    await this.makeRequestAsyncOmit({ method, bucketName, query, headers }, payload)
+  }
+
+  async removeBucketLifecycle(bucketName: string): Promise<void> {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    const method = 'DELETE'
+    const query = 'lifecycle'
+    await this.makeRequestAsyncOmit({ method, bucketName, query }, '', [204])
+  }
+
+  async setBucketLifecycle(bucketName: string, lifeCycleConfig: LifecycleConfig | null): Promise<void> {
```

Comment:
Would we need to add the empty string to the type of `lifeCycleConfig` as an empty string is also a possible argument for this parameter according to the [documentation](https://github.com/minio/minio-js/pull/1276/files#diff-b57590968a12cee85a37c1b91d8cc7092cd8b68e50b242c39a178121db82a797R646)?

```suggestion
  async setBucketLifecycle(bucketName: string, lifeCycleConfig: LifecycleConfig | null | ''): Promise<void> {
```